src/webview/js/js.js

```diff
@@ -426,8 +481,11 @@ const sendScrollMessage = () => {
     startMessageId: rangeHull.first,
     endMessageId: rangeHull.last,
   });
-  setMessagesReadAttributes(rangeHull);
-  prevMessageRange = messageRange;
+  // Only update things if we have visible + read messages.
+  if (messageRange.first < messageRange.last) {
+    setMessagesReadAttributes(rangeHull);
+    prevMessageRange = messageRange;
+  }
```

Comment:
This comment provides a "what" but I think a "why" would be helpful. We did after all just send a message to the RN side; why is it desirable not to make these updates? If I didn't remember the [chat thread](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M4526.2C.20unread.20bar.20jank/near/1144664) about this, I think I'd be quite mystified :slightly_smiling_face: 

In fact I think `setMessagesReadAttributes` should get called whenever we're sending that message (or "webview outbound event" -- "message" is a confusing word to use here, sorry). The event causes our RN code to go tell the server to please mark those messages as read, while `setMessagesReadAttributes` marks them immediately as read in the CSS, causing the green bar to fade away. If in some conditions one of those gets called but not the other, that makes it easy to have a bug where some message gets marked as read in one place but not the other.

Reply:
Done. I wasn't sure whether to include the `setMessagesReadAttributes`/`sendMessage` in the if statement — i opted not to, since the `sendMessage` seems like it could possible be important even if we don't update the `prevMessageRange`, but I'm open to doing it the other way if that seems better.

Reply:
Yep, this way is right! In fact there are two things we do in the outer code when we get one of these outbound events -- here's from `handleWebViewOutboundEvent.js` (which is also the file that defines these types like WebViewOutboundEventScroll`:
```js
    case 'scroll':
      fetchMore(props, event);
      markRead(props, event);
      break;
```

The latter of those two functions is telling the server to mark things as read, so it'll do nothing given an empty range like this (or in general if there were no still-unread messages in the range.) But the former is important for fetching more messages if the user has scrolled near the edge of what we have so far.